File: GoF_Design_Pattern/Adapter_Pattern/Adapter_with_Extends/java/PrintBanner.java
package Software_Engineering.GoF_Design_Pattern.Adapter_Pattern.Adapter_with_Extends.java;


import Software_Engineering.GoF_Design_Pattern.Adapter_Pattern.Adapter_with_Extends.Banner;
import Software_Engineering.GoF_Design_Pattern.Adapter_Pattern.Adapter_with_Extends.Print;

public class PrintBanner extends Banner implements Print {

    public PrintBanner(String string) {
        super(string);
    }

    @Override
    public void printWeak() {
        showWithParen();
    }

    @Override
    public void printStrong() {
        showWithAster();
    }

}

File: GoF_Design_Pattern/Adapter_Pattern/Adapter_with_Extends/python/PrintBanner.py
#! /etc/anaconda/python3

from Print import Print
from Banner import Banner


class PrintBanner(Banner, Print):

    def __init__(self, string):
        super(PrintBanner, self).__init__(string)

    def print_weak(self):
        self.show_with_paten()

    def print_strong(self):
        self.show_with_aster()
